app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,32 @@
+# responsible for creating and deleting user's favorite tracks
+# depends on asset_id
+# user must be logged_in
+# when favorite track is created it triggers favorites_count counter cache on asset
+class FavoritesController < ApplicationController
+  before_action :find_user
+  before_action :find_asset
+  before_action :require_login
+  before_action :find_favorite_track, only: :destroy
+
+  def create
+    raise ArgumentError unless @asset
```

Comment:
This was legacy from `toggle_favorites` controller method. Not sure if it'd still be a problem for this implementation, but wanted to catch any in Bugsnag for investigation.
```
    return false unless logged_in? && asset # no bullshit	
```